src/audio/mux/mux.c

```diff
@@ -213,11 +213,9 @@ static uint8_t get_stream_index(struct comp_data *cd, uint32_t pipe_id)
 {
 	int i;
 
-	for (i = 0; i < MUX_MAX_STREAMS; i++) {
-		comp_cl_err(&comp_mux, "got %u", cd->config.streams[i].pipeline_id);
+	for (i = 0; i < MUX_MAX_STREAMS; i++)
 		if (cd->config.streams[i].pipeline_id == pipe_id)
 			return i;
-	}
```

Comment:
@keyonjie the change looks good. But I wonder if the return 0 at the of the function is correct? If a match is not found, we should return a negative value isn't it?

Reply:
> @keyonjie the change looks good. But I wonder if the return 0 at the of the function is correct? If a match is not found, we should return a negative value isn't it?

That's a good point. I think the intention of the original author just wanted to return and use the first index (0) if the matched one not found, please check in the caller side, they don't check the return value and just use it as index blindly.

Reply:
Actually this is unsafe as it's used for an array lookup and may assign a source/sink for usage that is not in the correct state (and may crash etc). Can we check the result and return an error here (and handle it in the caller.)

Reply:
> Actually this is unsafe as it's used for an array lookup and may assign a source/sink for usage that is not in the correct state (and may crash etc). Can we check the result and return an error here (and handle it in the caller.)

Sure, let me add a commit to address this.